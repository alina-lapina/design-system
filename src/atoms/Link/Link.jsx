import React from 'react';
import PropTypes from 'prop-types';

const Link = ({
	children, href, icon, isExternal, linkType,
}) => (
	<div className="link">
		<div className="index">
			<a
				className={`link-text ${linkType} ${icon && 'with-icon'}`}
				href={href}
				target={isExternal ? '_blank' : ''}
				rel={isExternal ? 'noopener noreferrer' : ''}
			>{icon}<span className="link-text">{children}</span>
			</a>
		</div>
		<div className="animate-background" />
	</div>
);

Link.defaultProps = {
	isExternal: false,
	linkType: 'regular',
};

Link.propTypes = {
	children: PropTypes.node.isRequired,
	href: PropTypes.string.isRequired,
	icon: PropTypes.element,
	isExternal: PropTypes.bool,
	linkType: PropTypes.oneOf([
		'regular',
		'profiled',
		'header',
	]),
};

export default Link;

/*
options:
	href: required string for url
	icon: optional element for displaying a left side icon
	isExternal: optional boolean which makes the link open in new tab
	linkType: optional styling profiles
	text: required string for text
 */
