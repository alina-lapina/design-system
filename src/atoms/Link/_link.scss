$profiled-font-size: 18px;
$header-font-size: 18px;

.link {
  cursor: pointer;
  display: inline-block;
  position: relative;

  .link-text {
    transition-delay: .1s;

    &.with-icon {
      color: $ssb-dark-6;
      text-decoration: none;
    }
  }

  .regular {
    color: $ssb-green-4;

  }
  .profiled {
    color: $ssb-green-4;
    font-size: $profiled-font-size;
    font-weight: bold;
    transition-delay: .1s;
  }
  .header {
    color: $ssb-dark-6;
    font-size: $header-font-size;
    text-decoration-color: $ssb-green-4;
  }

  .animate-background {
    bottom: 0;
    background: $ssb-green-4;
    height: 0;
    left: -2px;
    position: absolute;
    transition: height .2s;
    width: calc(100% + 4px);
    z-index: -1;
  }

  &:hover {
    .animate-background {
      height: 100%;
    }
    .link-text {
      color: $ssb-white;
    }
  }
}