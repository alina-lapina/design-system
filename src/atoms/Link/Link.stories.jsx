import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Link from './Link';

storiesOf('Link', module).addDecorator(centered)
	.add('Regular', () => (
		<div>
			<Link href=" " text="A regular link" />
			<br />
			<Link href=" " text="A regular link with icon" icon={<span>&gt;</span>}/>
		</div>
	))
	.add('Profiled', () => (
		<div>
			<Link href=" " text="A profiled link" linkType="profiled" />
			<br />
			<Link href=" " text="A profiled link with icon" linkType="profiled" icon={<span>&gt;</span>}/>
		</div>
	))
	.add('Header', () => (
		<div>
			<Link href=" " text="A header link" linkType="header" />
			<br />
			<Link href=" " text="A header link with icon" linkType="header" icon={<span>&gt;</span>}/>
		</div>
	));
