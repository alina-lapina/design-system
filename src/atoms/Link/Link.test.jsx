import React from 'react';
import TestRenderer from 'react-test-renderer';
import Link from './Link';

describe('Link component', () => {
	test('Matches the snapshot', () => {
		const button = TestRenderer.create(<Link text="A link" href=" " />);
		expect(button.toJSON()).toMatchSnapshot();
	});
});
