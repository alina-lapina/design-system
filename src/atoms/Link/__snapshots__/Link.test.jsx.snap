// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component Matches the snapshot 1`] = `
<div
  className="link"
>
  <a
    className="link-text regular undefined"
    href=" "
  >
    <span
      className="link-text"
    >
      A link
    </span>
  </a>
  <div
    className="animate-background"
  />
</div>
`;
